test/quantization/fx/test_quantize_fx.py

```diff
@@ -5975,6 +5975,118 @@ def forward(self, x):
             }
             self.checkGraphModuleNodes(quantized_model, expected_node_occurrence=node_occurrence)
 
+    def test_mixed_dtypes(self):
+        """
+        Test that multiple dtypes can be used in the same model for different layers,
+        and the dtypes will be converted correctly between the layers.
+        """
+        class MyModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear1 = torch.nn.Linear(5, 5)
+                self.linear2 = torch.nn.Linear(5, 5)
+                self.sigmoid = torch.nn.Sigmoid()
+                self.tanh = torch.nn.Tanh()
+                self.float_functional = torch.ao.nn.quantized.FloatFunctional()
+
+            def forward(self, x: torch.Tensor):
+                x = self.linear1(x)  # qint32
+                x = self.linear2(x)  # quint8
+                linear2 = x
+                x = self.sigmoid(x)  # back to qint32
+                x = self.tanh(x)  # back to quint8
+                x = self.float_functional.add(linear2, x)  # adding two quint8's together
+                return x
+
+        def make_qconfig(scale, zp, dtype):
+            return QConfig(
+                activation=FixedQParamsObserver.with_args(scale=scale, zero_point=zp, dtype=dtype),
+                weight=torch.ao.quantization.default_weight_observer)
+
+        # Set up a QConfigMapping that specifies different qparams and dtypes for different layers
+        qconfig_mapping = QConfigMapping() \
+            .set_global(get_default_qconfig("qnnpack")) \
+            .set_module_name("linear1", make_qconfig(1234, 11, torch.qint32)) \
+            .set_module_name("linear2", make_qconfig(2345, 22, torch.quint8)) \
+            .set_object_type(torch.nn.Sigmoid, make_qconfig(3456, 33, torch.qint32)) \
+            .set_object_type(torch.nn.Tanh, make_qconfig(4567, 44, torch.quint8))
+
+        # Set up BackendConfig that supports the dtypes configured in the above QConfigMapping
+        weighted_op_qint32_dtype_config = DTypeConfig(
+            input_dtype=torch.qint32,
+            output_dtype=torch.qint32,
+            weight_dtype=torch.qint8,
+            bias_dtype=torch.float,
+        )
+        fixed_qparams_op_quint8_dtype_config = DTypeConfig(
+            input_dtype=torch.quint8,
+            output_dtype=torch.quint8,
+        )
+        fixed_qparams_op_qint32_dtype_config = DTypeConfig(
+            input_dtype=torch.qint32,
+            output_dtype=torch.qint32,
+        )
+        backend_config = get_qnnpack_backend_config()
+        for config in backend_config.configs:
+            if config.pattern == torch.nn.Linear:
+                config.add_dtype_config(weighted_op_qint32_dtype_config)
+            elif config.pattern in [torch.nn.Sigmoid, torch.nn.Tanh]:
```

Comment:
double checking, do you mean to add both when you see either Sigmoid and Tanh? or to add them respectively.

Reply:
Yeah right now this supports both dtypes for both ops. I guess it can be more specific but I think this is also fine